File: 102-2_iOS程式設計/MediaSample/MediaSample/ViewController.m
//
//  ViewController.m
//  MediaSample
//
//  Created by CSIEVNUT403 on 2014/5/1.
//  Copyright (c) 2014年 CSIEVNUT403. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

@synthesize playBtn, stopBtn, audioPlayer;

- (IBAction)play:(id)sender {
    self.audioPlayer.currentTime = 0;
    [self.audioPlayer play];
    //self.audioPlayer.play;
}

- (IBAction)stop:(id)sender {
    [self.audioPlayer stop];
}

- (void)viewDidLoad
{
    [super viewDidLoad];
	// Do any additional setup after loading the view, typically from a nib.
    
    NSString *filePath = [[NSBundle mainBundle] pathForResource:@"music" ofType:@"m4a"];
    NSURL *fileURL = [[NSURL alloc] initFileURLWithPath:filePath];
    
    self.audioPlayer = [[AVAudioPlayer alloc] initWithContentsOfURL:fileURL error:nil];
    [self.audioPlayer prepareToPlay];
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: 105-2_iOS程式設計/ImageView03/ImageView03/ViewController.swift
//
//  ViewController.swift
//  ImageView03
//
//  Created by mojado on 2017/4/6.
//  Copyright © 2017年 VNU. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var imageShow: UIImageView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        imageShow.image = UIImage(named: "img01")
    }

    @IBAction func borderClick(sender: UIButton) {
        imageShow.layer.borderColor = UIColor.redColor().CGColor
        imageShow.layer.borderWidth = 5
    }
    
    @IBAction func roundClick(sender: UIButton) {
        imageShow.layer.cornerRadius = 30
        imageShow.layer.masksToBounds = true
    }
    
    @IBAction func shadowClick(sender: UIButton) {
        imageShow.layer.shadowColor = UIColor.blueColor().CGColor
        imageShow.layer.shadowOffset = CGSizeMake(10, 10)
        imageShow.layer.shadowOpacity = 0.8
        imageShow.layer.shadowRadius = 5
    }
    
    @IBAction func normalClick(sender: UIButton) {
        imageShow.layer.borderWidth = 0
        imageShow.layer.cornerRadius = 0
        imageShow.layer.masksToBounds = false
        imageShow.layer.shadowOpacity = 0
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

}

